lib/setup_functions.php

```diff
@@ -0,0 +1,965 @@
+<?php
+
+function checkInFile($aCMDResult)
+{
+    if ($aCMDResult['import-data'] || $aCMDResult['all']) {
+        if (!isset($aCMDResult['osm-file'])) {
+            fail('missing --osm-file for data import');
+        }
+
+        if (!file_exists($aCMDResult['osm-file'])) {
+            fail('the path supplied to --osm-file does not exist');
+        }
+
+        if (!is_readable($aCMDResult['osm-file'])) {
+            fail('osm-file "'.$aCMDResult['osm-file'].'" not readable');
+        }
+    }
+}
+
+function prepSystem($aCMDResult)
+{
+    // by default, use all but one processor, but never more than 15.
+    $iInstances = isset($aCMDResult['threads'])
+                ? $aCMDResult['threads']
+                : (min(16, getProcessorCount()) - 1);
+
+    if ($iInstances < 1) {
+        $iInstances = 1;
+        warn("resetting threads to $iInstances");
+    }
+
+    // Assume we can steal all the cache memory in the box (unless told otherwise)
+    if (isset($aCMDResult['osm2pgsql-cache'])) {
+        $iCacheMemory = $aCMDResult['osm2pgsql-cache'];
+    } else {
+        $iCacheMemory = getCacheMemoryMB();
+    }
+
+    $sModulePath = CONST_Database_Module_Path;
+    info('module path: ' . $sModulePath);
+
+    return array($iCacheMemory,$iInstances);
+}
+
+function prepDB($aCMDResult)
+{
+    $sModulePath = CONST_Database_Module_Path;
+    $aDSNInfo = DB::parseDSN(CONST_Database_DSN);
+    if (!isset($aDSNInfo['port']) || !$aDSNInfo['port']) $aDSNInfo['port'] = 5432;
+    
+    if ($aCMDResult['create-db'] || $aCMDResult['all']) {
+        info('Create DB');
+        $bDidSomething = true;
+        $oDB = DB::connect(CONST_Database_DSN, false);
+        if (!PEAR::isError($oDB)) {
+            fail('database already exists ('.CONST_Database_DSN.')');
+        }
+    
+        $sCreateDBCmd = 'createdb -E UTF-8 -p '.$aDSNInfo['port'].' '.$aDSNInfo['database'];
+        if (isset($aDSNInfo['username']) && $aDSNInfo['username']) {
+            $sCreateDBCmd .= ' -U ' . $aDSNInfo['username'];
+        }
+        if (isset($aDSNInfo['hostspec']) && $aDSNInfo['hostspec']) {
+            $sCreateDBCmd .= ' -h ' . $aDSNInfo['hostspec'];
+        }
+    
+        $aProcEnv = null;
+        if (isset($aDSNInfo['password']) && $aDSNInfo['password']) {
+            $aProcEnv = array_merge(array('PGPASSWORD' => $aDSNInfo['password']), $_ENV);
+        }
+    
+        $result = runWithEnv($sCreateDBCmd, $aProcEnv);
+        if ($result != 0) fail('Error executing external command: '.$sCreateDBCmd);
+    }
+    
+    if ($aCMDResult['setup-db'] || $aCMDResult['all']) {
```

Comment:
create-db and setup-db should still be provided in separate functions. It makes the code easier to read.